packages/js/src/ui/components/elements/Preferences/Preferences.tsx

```diff
@@ -40,178 +38,94 @@ export const Preferences = () => {
     }));
   });
 
-  const optimisticUpdate =
-    (preference?: Preference) =>
-    async ({ channel, enabled }: { channel: ChannelType; enabled: boolean }) => {
-      await preference?.update({
-        channels: {
-          [channel]: enabled,
-        },
-      });
-    };
+  const updatePreference = (preference?: Preference) => async (channels: ChannelPreference) => {
+    await preference?.update({
+      channels,
+    });
+  };
+
+  const bulkUpdatePreferences = (preferences: Preference[]) => async (channels: ChannelPreference) => {
+    await novu.preferences.bulkUpdate(
+      preferences.map((el) => {
+        const oldChannels = Object.keys(el.channels);
+        const channelsToUpdate = Object.keys(channels)
+          .filter((channel) => oldChannels.includes(channel))
+          .reduce((acc, channel) => {
+            acc[channel as keyof ChannelPreference] = channels[channel as keyof ChannelPreference];
+
+            return acc;
+          }, {} as ChannelPreference);
+
+        return { preference: el, channels: channelsToUpdate };
+      })
+    );
+  };
```

Comment:
bulk update preferences, but only update the channel preferences that workflow includes (do not update not existing as it results in UI glitch)